crates/ironrdp-server/src/server.rs

```diff
@@ -0,0 +1,292 @@
+use std::{io::Cursor, net::SocketAddr};
+
+use anyhow::Result;
+use bytes::BytesMut;
+use tokio::net::TcpListener;
+
+use ironrdp_acceptor::{self, Acceptor, BeginResult};
+use ironrdp_pdu::{
+    self,
+    input::{
+        fast_path::{FastPathInput, FastPathInputEvent},
+        InputEventPdu,
+    },
+    mcs, nego, rdp, Action, PduParsing,
+};
+use ironrdp_tokio::{Framed, FramedRead, FramedWrite, TokioFramed};
+use tokio_rustls::TlsAcceptor;
+
+use crate::{
+    builder, capabilities,
+    display::{DisplayUpdate, RdpServerDisplay},
+    encoder::UpdateEncoder,
+    handler::RdpServerInputHandler,
+};
+
+#[derive(Clone)]
+pub struct RdpServerOptions {
+    pub addr: SocketAddr,
+    pub security: RdpServerSecurity,
+}
+
+#[derive(Clone)]
+pub enum RdpServerSecurity {
+    None,
+    TLS(TlsAcceptor),
```

Comment:
Nit: the convention in Rust is to count acronyms and contractions of compound words as one word so this should be `Tls` rather than `TLS`

```suggestion
    Tls(TlsAcceptor),
```